### ISSUE TITLE
Export recipe and calculation 

### ISSUE THREAD

**Issue by Tirguy** [nextcloud/cookbook#1665]
Some ideas like that... that i would like to have in the application : 

- Possibility to export one or more recipe in PDF / ODT / DOCX
- Possibility to export one or more recipe... or one or more recipe group for another cookbook application
- Possibility to share one recipe on a public link
- Calculation of the calories of a dish

Thank's 
Tirguy

**Comment by christianlupus** [comment]
Hello and thanks for the ideas. Let me go through them one at a time.

> - Possibility to export one or more recipe in PDF / ODT / DOCX

I will link to the issues #286 #1241 #1485.

To be precise, as far as I know, there is not yet a request for DOCX. I would avoid that though for multiple reasons:
1. DOCX is a proprietary format. It is not easy to generate that.
2. I fear that soon after, the request for an import feature will arise. This is definitely out of scope.

> - Possibility to export one or more recipe... or one or more recipe group for another cookbook application

What do you mean by *export*? Do you mean a dedicated format like in #831?

> - Possibility to share one recipe on a public link

Again, there are already some issues: #1485 #392 #120

> - Calculation of the calories of a dish

If there is no information about nutrition available, this seems rather impossible to me.
One could use the serving number and the contained energy from the nutrition data to break it down per serving. Is that what you are thinking of?

**Comment by Tirguy** [comment]
Thank you Christian for this feedback... Here are my answers:

```
To be precise, as far as I know, there is not yet a request for DOCX. I would avoid that though for multiple reasons:
  1 - DOCX is a proprietary format. It is not easy to generate that.
  2 - I fear that soon after, the request for an import feature will arise. This is definitely out of scope

```
OK i understand... And it's true that someone could ask the ability to import .doc(x) and it's not really possible ;-) but it could be nice to export a recipe in a PDF format :-D 

Note : I've used this library (https://github.com/PHPOffice/PHPWord) to generate .docx in one of my php project

```
Possibility to export one or more recipe... or one or more recipe group for another cookbook application
- What do you mean by export? Do you mean a dedicated format like in #831 ?
Possibility to share one recipe on a public link
- Again, there are already some issues: #1485 #392 #120
```

In fact, there are two ideas:
The first is to be able to exchange recipes from one Cookbook to another through an exchange of a link or a file for example which would be exported and then imported into the second Cookbook. This idea is a bit like what appears in the other requests you mention and which I had not paid attention to (I thought I was the first!!!).
The second is to be able to share a recipe link without the person having an account on my Nextclood server. This is a bit like file sharing, but in the form of access to a single recipe page.

In short, all my remarks are more or less similar to the requests you already had. But in general these requests go in several directions and not to the point. Personally, being able to generate a zip archive that integrates an export of the recipe in JSON format + its image and then being able to import it into another Cookbook would already be great!

For the last idea, yes, that's what I'm thinking of... ;-) even if it might take a few lines of code to provide the option.

**Comment by datavizzard** [comment]
I think it is necessary to be able to export the recipes, just because of the backup. A useful export function in PDF or txt/csv to share or print them would be very welcome.

**Comment by datavizzard** [comment]
If everything goes wrong, it would be good if the recipes are structured as a PDF in the core directory, especially for simple users. This can be easily shared, saved or passed on.

**Comment by Markoise** [comment]
An export function would actually be a great thing. An export to an ODT file would be better than PDF if DOCX is not desired due to the proprietary background.

This would allow you to share recipes with another person who doesn't use the cookbook app or Nextcloud. 

Even in the event that you no longer use Nextcloud, you would then have a "backup" of the recipes that you can process yourself. JSON or PDF are not the right choice for everyone, I think.
